connection.js

```diff
@@ -682,282 +626,141 @@ function conn_log() {
     }
 }
 
-function beforeRankedUnrankedGamesSpecial(before, extra, argNameString, special) {
-    const isExtra = (extra !== "");
-    if (argNameString.includes("unranked")) {
-        return `${before}${extra}unranked games`; //ex: "for blitz unranked games"
-    } else if (argNameString.includes("ranked")) {
-        return `${before}${extra}ranked games`;   //ex: "for ranked games"
-    } else if (isExtra) {
-        return `${before}${extra}games`           //ex: "for correspondence games"
-    } else if (special !== "") {
-        return `${before}${special}games`         //ex: "from all games"
-    } else {
-        return "";
-    }
+function generate_r_u_strings_connection(rankedSetting, speedSetting) {
+    const r_u = rankedSetting ? "unranked" : "ranked";
+    return { r_u,
+             for_r_u_games: `for ${r_u} games`,
+             for_blc_r_u_games: `for ${speedSetting} ${r_u} games` };
 }
 
-function rankToString(r) {
-    const R = Math.floor(r);
-    if (R >= 30)  return (R-30+1) + 'd'; // R>=30 : 1 dan or stronger
-    else          return (30-R) + 'k';   // R<30  : 1 kyu or weaker
-}
-
-function bannedFamilyReject(argNameString, uid, notificationUid) {
-    const rankedUnranked = beforeRankedUnrankedGamesSpecial("from ", "", argNameString, "all ");
-    conn_log(`${uid} ${notificationUid} is banned ${rankedUnranked}`);
-    return { reject: true, msg: `You (${uid} ${notificationUid}) are banned ${rankedUnranked} on this bot by bot admin, you may try changing the ranked/unranked setting` };
+function checkUserIsInSet(user, currSet) {
+    return (currSet.has(user.id) || currSet.has(user.username));
 }
 
-function noAutohandicapReject(argNameString) {
-    const rankedUnranked = beforeRankedUnrankedGamesSpecial("for ", "", argNameString, "");
-    conn_log(`no autohandicap ${rankedUnranked}`);
-    return { reject: true, msg: `For easier bot management, -automatic- handicap is disabled on this bot ${rankedUnranked}, please manually select the number of handicap stones you want in -custom handicap-, for example 2 handicap stones, you may try changing the ranked/unranked setting` };
+function getBlackWhiteListReject(setName, for_r_u_games) {
+    conn_log(`user is not allowed ${for_r_u_games} (${setName})`);
+    return { reject: true, msg: `This bot uses a ${setName}: your user name/id `
+                                + `is not allowed on this bot ${for_r_u_games}, `
```

Comment:
I would prefer "This bot uses a ${setName}: You are not allowed on this bot ${for_r_u_games}". mentioning `name/id` gives out unnecessarily info to malicious users, and plain confusing info for others.